### PR TITLE
Expose traces jsonnet configuration

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Follow up for #199

### CODE REVIEW

File: jsonnet/lib/observatorium-api.libsonnet [hunk 1]
```diff
@@ -127,6 +128,13 @@ function(params) {
                     '--logs.tail.endpoint=' + api.config.logs.tailEndpoint,
                     '--logs.write.endpoint=' + api.config.logs.writeEndpoint,
                   ] else []
+              ) + (
```
Comment (ianbillett) [inline jsonnet/lib/observatorium-api.libsonnet:131]: duplicated `) + (`

Comment (pavolloffay) [reply jsonnet/lib/observatorium-api.libsonnet:132]: yap, fixed

File: jsonnet/lib/observatorium-api.libsonnet [hunk 2]
```diff
@@ -127,6 +129,12 @@ function(params) {
...
+                if api.config.traces != {} then
+                  [
+                    '--traces.write.endpoint=' + api.config.traces.writeEndpoint,
```
Comment (ianbillett) [inline jsonnet/lib/observatorium-api.libsonnet:135]: Are we missing a `--traces.read.endpoint` too?

Comment (ianbillett) [reply jsonnet/lib/observatorium-api.libsonnet:135]: Ah - there's not a flag for that in the API [yet](https://github.com/observatorium/api/blob/b3a191802b501fb6051e73f2f986240e1e406be4/main.go#L864-L865)

Comment (pavolloffay) [reply jsonnet/lib/observatorium-api.libsonnet:135]: that hasn't been added yet (https://github.com/observatorium/api/pull/199)

@esnible is working on it.

**Comment by pavolloffay** [comment]
Can somebody restart the CI?

```
--- PASS: TestTenantsRetryAuthenticationProviderRegistration (18.92s)
    --- PASS: TestTenantsRetryAuthenticationProviderRegistration/tenants-authenticator-provider-retry (5.76s)
=== CONT  TestRulesAPI
    services.go:57: services.go:57:
        
         unexpected error: the service rules-minio is not ready: Get "http://127.0.0.1:49196/minio/health/live": dial tcp 127.0.0.1:49196: connect: connection refused
```

**Comment by pavolloffay** [comment]
There seems to be another flaky test

```
=== CONT  TestRulesAPI
    services.go:57: services.go:57:
        
         unexpected error: the service rules-minio is not ready: Get "http://127.0.0.1:49170/minio/health/live": dial tcp 127.0.0.1:49170: connect: connection refused
        
16:32:24 Killing rules-minio
16:32:24 Error response from daemon: Cannot kill container: e2e_rules_api-rules-minio: No such container: e2e_rules_api-rules-minio

16:32:24 Unable to kill service rules-minio : exit status 1
16:32:24 Killing opa
16:32:24 Killing gubernator
16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="Gubernator 1.0.0-rc.8 (amd64/linux)"
16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=warning msg="while deserialize member-list peer" category=gubernator error="error decoding peer: EOF"
16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="GRPC Listening on 0.0.0.0:8881 ..." category=gubernator
16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="2022/03/11 16:32:24 [DEBUG] memberlist: Stream connection from=127.0.0.1:58388" category=gubernator
16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="2022/03/11 16:32:24 [DEBUG] memberlist: Initiating push/pull sync with:  127.0.0.1:7946" category=gubernator
16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="HTTP Gateway Listening on 0.0.0.0:8880 ..." category=gubernator
16:32:25 Ports for container e2e_logs_read_write_tail-gubernator >> Local ports: map[grpc:8881 http:8880] Ports available from host: map[grpc:49189 http:49190]
16:32:25 Starting opa
16:32:25 Killing dex
16:32:26 opa: {"addrs":[":8181"],"diagnostic-addrs":[],"level":"info","msg":"Initializing server.","time":"2022-03-11T16:32:26Z"}
16:32:26 Ports for container e2e_logs_read_write_tail-opa >> Local ports: map[http:8181] Ports available from host: map[http:49191]
--- FAIL: TestRulesAPI (51.45s)
```

**Comment by jessicalins** [comment]
> There seems to be another flaky test
> 
> ```
> === CONT  TestRulesAPI
>     services.go:57: services.go:57:
>         
>          unexpected error: the service rules-minio is not ready: Get "http://127.0.0.1:49170/minio/health/live": dial tcp 127.0.0.1:49170: connect: connection refused
>         
> 16:32:24 Killing rules-minio
> 16:32:24 Error response from daemon: Cannot kill container: e2e_rules_api-rules-minio: No such container: e2e_rules_api-rules-minio
> 
> 16:32:24 Unable to kill service rules-minio : exit status 1
> 16:32:24 Killing opa
> 16:32:24 Killing gubernator
> 16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="Gubernator 1.0.0-rc.8 (amd64/linux)"
> 16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=warning msg="while deserialize member-list peer" category=gubernator error="error decoding peer: EOF"
> 16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="GRPC Listening on 0.0.0.0:8881 ..." category=gubernator
> 16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="2022/03/11 16:32:24 [DEBUG] memberlist: Stream connection from=127.0.0.1:58388" category=gubernator
> 16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="2022/03/11 16:32:24 [DEBUG] memberlist: Initiating push/pull sync with:  127.0.0.1:7946" category=gubernator
> 16:32:24 gubernator: time="2022-03-11T16:32:24Z" level=info msg="HTTP Gateway Listening on 0.0.0.0:8880 ..." category=gubernator
> 16:32:25 Ports for container e2e_logs_read_write_tail-gubernator >> Local ports: map[grpc:8881 http:8880] Ports available from host: map[grpc:49189 http:49190]
> 16:32:25 Starting opa
> 16:32:25 Killing dex
> 16:32:26 opa: {"addrs":[":8181"],"diagnostic-addrs":[],"level":"info","msg":"Initializing server.","time":"2022-03-11T16:32:26Z"}
> 16:32:26 Ports for container e2e_logs_read_write_tail-opa >> Local ports: map[http:8181] Ports available from host: map[http:49191]
> --- FAIL: TestRulesAPI (51.45s)
> ```

related to https://github.com/efficientgo/e2e/issues/11 
I'll open an issue for it
For now I've rerun CI :)

**Comment by pavolloffay** [comment]
@jessicalins it seems that the CI failed again 

```
=== CONT  TestRulesAPI
    services.go:57: services.go:57:
        
         unexpected error: the service rules-minio is not ready: Get "http://127.0.0.1:49166/minio/health/live": dial tcp 127.0.0.1:49166: connect: connection refused
        
10:14:01 Killing rules-minio
10:14:01 Killing opa
10:14:01 Error response from daemon: Cannot kill container: e2e_rules_api-rules-minio: No such container: e2e_rules_api-rules-minio
```

**Comment by jessicalins** [comment]
@pavolloffay could you update your branch with latest main? this [pr](https://github.com/observatorium/api/pull/251) should have fixed CI

**Comment by pavolloffay** [comment]
CI passed now :)

**Review by jessicalins — APPROVED** [review]
LGTM, I've re-requested @ianbillett 's review as well

**Review by bwplotka — APPROVED** [review]
👍🏽

**Pull request merged**